feat : 공연 추가

beetrbgus opened the pull request:
## 📃 관련 이슈
- #51 

## ✨ 변경 내용
- 변경 사항 간단 요약

## 🔎 참고 사항 (선택)
- 추가 설명, 참고 링크 등

beetrbgus committed:
feat : 작업 내용 커밋

# #############################
# # 타입 설정                  ##
# #############################

# # feat: 기능 구현
# # refactor: 코드 구조 개선, 변수명 변경 등 코드 리팩토링
# # fix: 버그 또는 오류 해결
# # style: CSS 등 스타일 관련 변경
# # test: 테스트 코드 추가 또는 수정
# # docs: README.md 등 문서 관련 변경
# # chore: 파일 또는 디렉토리 구조 변경, 패키지 설치 등 빌드 시스템이나 외부 도구 관련 작업
# # build: CI/CD 관련 변경

beetrbgus committed:
feat : 작업 내용 추가

beetrbgus committed:
Merge branch 'develop' into feat/51

beetrbgus committed:
feat : 검증 요소 추가 및 생성로직 수정

# feat: 기능 구현
# refactor: 코드 구조 개선, 변수명 변경 등 코드 리팩토링
# fix: 버그 또는 오류 해결
# style: CSS 등 스타일 관련 변경
# test: 테스트 코드 추가 또는 수정
# docs: README.md 등 문서 관련 변경
# chore: 파일 또는 디렉토리 구조 변경, 패키지 설치 등 빌드 시스템이나 외부 도구 관련 작업
# build: CI/CD 관련 변경

beetrbgus committed:
Merge remote-tracking branch 'origin/feat/51' into feat/51

beetrbgus committed:
feat : 회차 정보 추가

# feat: 기능 구현
# refactor: 코드 구조 개선, 변수명 변경 등 코드 리팩토링
# fix: 버그 또는 오류 해결
# style: CSS 등 스타일 관련 변경
# test: 테스트 코드 추가 또는 수정
# docs: README.md 등 문서 관련 변경
# chore: 파일 또는 디렉토리 구조 변경, 패키지 설치 등 빌드 시스템이나 외부 도구 관련 작업
# build: CI/CD 관련 변경

beetrbgus committed:
Merge branch 'develop' of https://github.com/show-hive/showhive-backend into feat/51

# Conflicts:
#	showhive-boot/admin/src/main/java/com/showhive/admin/application/command/usecase/performance/impl/CreatePerformanceUseCaseImpl.java
#	showhive-domain/domain/src/main/java/com/showhive/category/repository/query/CategoryQueryRepository.java
#	showhive-domain/domain/src/main/java/com/showhive/performance/domain/PerformanceSession.java
#	showhive-domain/domain/src/main/java/com/showhive/performance/repository/command/PerformanceCommandRepository.java
#	showhive-domain/domain/src/main/java/com/showhive/venue/repository/query/SeatQueryRepository.java
#	showhive-infra/infra-persistence/src/main/java/com/showhive/category/query/CategoryQueryJpaRepository.java
#	showhive-infra/infra-persistence/src/main/java/com/showhive/performance/PerformanceCommandRepositoryImpl.java
#	showhive-infra/infra-persistence/src/main/java/com/showhive/performance/command/PerformanceSessionCommandJpaRepository.java
#	showhive-infra/infra-persistence/src/main/java/com/showhive/performance/query/PerformanceCategoryMapQueryJpaRepository.java
#	showhive-infra/infra-persistence/src/main/java/com/showhive/venue/SeatQueryRepositoryImpl.java
#	showhive-infra/infra-persistence/src/main/java/com/showhive/venue/query/SeatQueryJpaRepository.java

beetrbgus committed:
Merge branch 'refs/heads/develop' into feat/51

Jiihyun commented on showhive-boot/admin/src/main/java/com/showhive/admin/interfaces/category/dto/CreateSessionRequest.java:
```diff
@@ -0,0 +1,66 @@
+        @NoArgsConstructor
+        @AllArgsConstructor
+        @Schema(description = "공연 회차 세부 정보")
+        public static class SessionDto {
```
C: 여기는 왜 dto인가요?!

Jiihyun commented on showhive-domain/domain/src/main/java/com/showhive/performance/entity/Performance.java:
```diff
@@ -64,6 +79,23 @@ public static Performance create(String title, Long venueId, Long runningTime, S
+        CANCELED("주최 측 사정으로 공연 취소 됨"),
+        POSTPONED("공연이 연기"),
+        ARCHIVED("끝난 공연을 이력 보관용으로 유지"),
+        ;
```
C: 카테고리 경계가 다소 애매한 것 같아요! 뭔가 중복이 가능한 느낌??
- FINISHED랑 CLOSED의 차이가 무엇일까요??
- 공연이 끝나면 FINISHED랑 ARCHIVED 둘 다 허용되지 않나요??!

Jiihyun commented on showhive-infra/infra-persistence/src/main/java/com/showhive/performance/query/PerformanceCategoryMapQueryJpaRepository.java:
```diff
@@ -1,7 +1,14 @@
+
+    PerformanceCategoryMap findByCategory_Id(Long categoryCategoryId);
+
+    List<PerformanceCategoryMap> findAllByCategory_IdIn(Collection<Long> categoryIds);
```
C: `_`를 포함한 이유가 있나여?!

Jiihyun reviewed (commented):
고생하셨습니다!!!
리뷰가 늦어 죄송해요🥲
로직상 의문이 드는 곳은 없었고, 컨벤션이나 enum 관련해서 몇가지 질문 남겨두었습니다!!

beetrbgus committed:
refactor : 오타 수정

# feat: 기능 구현
# refactor: 코드 구조 개선, 변수명 변경 등 코드 리팩토링
# fix: 버그 또는 오류 해결
# style: CSS 등 스타일 관련 변경
# test: 테스트 코드 추가 또는 수정
# docs: README.md 등 문서 관련 변경
# chore: 파일 또는 디렉토리 구조 변경, 패키지 설치 등 빌드 시스템이나 외부 도구 관련 작업
# build: CI/CD 관련 변경

beetrbgus committed:
refactor : 오타 수정

# feat: 기능 구현
# refactor: 코드 구조 개선, 변수명 변경 등 코드 리팩토링
# fix: 버그 또는 오류 해결
# style: CSS 등 스타일 관련 변경
# test: 테스트 코드 추가 또는 수정
# docs: README.md 등 문서 관련 변경
# chore: 파일 또는 디렉토리 구조 변경, 패키지 설치 등 빌드 시스템이나 외부 도구 관련 작업
# build: CI/CD 관련 변경

beetrbgus committed:
refactor : Entity 네임 변경

# feat: 기능 구현
# refactor: 코드 구조 개선, 변수명 변경 등 코드 리팩토링
# fix: 버그 또는 오류 해결
# style: CSS 등 스타일 관련 변경
# test: 테스트 코드 추가 또는 수정
# docs: README.md 등 문서 관련 변경
# chore: 파일 또는 디렉토리 구조 변경, 패키지 설치 등 빌드 시스템이나 외부 도구 관련 작업
# build: CI/CD 관련 변경

beetrbgus committed:
refactor : Entity 네임 변경

# feat: 기능 구현
# refactor: 코드 구조 개선, 변수명 변경 등 코드 리팩토링
# fix: 버그 또는 오류 해결
# style: CSS 등 스타일 관련 변경
# test: 테스트 코드 추가 또는 수정
# docs: README.md 등 문서 관련 변경
# chore: 파일 또는 디렉토리 구조 변경, 패키지 설치 등 빌드 시스템이나 외부 도구 관련 작업
# build: CI/CD 관련 변경

beetrbgus committed:
fix : 테스트 추가
# feat: 기능 구현
# refactor: 코드 구조 개선, 변수명 변경 등 코드 리팩토링
# fix: 버그 또는 오류 해결
# style: CSS 등 스타일 관련 변경
# test: 테스트 코드 추가 또는 수정
# docs: README.md 등 문서 관련 변경
# chore: 파일 또는 디렉토리 구조 변경, 패키지 설치 등 빌드 시스템이나 외부 도구 관련 작업
# build: CI/CD 관련 변경

beetrbgus committed:
fix : 오류 수정
# feat: 기능 구현
# refactor: 코드 구조 개선, 변수명 변경 등 코드 리팩토링
# fix: 버그 또는 오류 해결
# style: CSS 등 스타일 관련 변경
# test: 테스트 코드 추가 또는 수정
# docs: README.md 등 문서 관련 변경
# chore: 파일 또는 디렉토리 구조 변경, 패키지 설치 등 빌드 시스템이나 외부 도구 관련 작업
# build: CI/CD 관련 변경

beetrbgus committed:
feat : Test 추가
# feat: 기능 구현
# refactor: 코드 구조 개선, 변수명 변경 등 코드 리팩토링
# fix: 버그 또는 오류 해결
# style: CSS 등 스타일 관련 변경
# test: 테스트 코드 추가 또는 수정
# docs: README.md 등 문서 관련 변경
# chore: 파일 또는 디렉토리 구조 변경, 패키지 설치 등 빌드 시스템이나 외부 도구 관련 작업
# build: CI/CD 관련 변경

beetrbgus committed:
feat : Test 추가
# feat: 기능 구현
# refactor: 코드 구조 개선, 변수명 변경 등 코드 리팩토링
# fix: 버그 또는 오류 해결
# style: CSS 등 스타일 관련 변경
# test: 테스트 코드 추가 또는 수정
# docs: README.md 등 문서 관련 변경
# chore: 파일 또는 디렉토리 구조 변경, 패키지 설치 등 빌드 시스템이나 외부 도구 관련 작업
# build: CI/CD 관련 변경
